Adding howto to set up french fiber provider SFR/RED

pgaultier opened the pull request:
Adding documentation for french provider SFR/RED
 * IPv4
 * IPv6
 * Phone

fabianfrz commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+In order to establish the IPv4 and IPv6 connection SFR/RED requires that the correct parameters are passed for the DHCP and DHCP6
+requests respectively
+
+select options DHCP and DHCPv6 in general configuration
```
Even with the dots it looks weird:
```suggestion
Select options DHCP and DHCPv6 in general configuration:
```

pgaultier replied:
Updated in last commit

fabianfrz commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+.. Note::
+    It is necessary to specify the following 'Request Options'
+
+* subnet-mask, broadcast-address, time-offset, routers, domain-name, domain-name-servers, host-name, ntp-servers, nis-domain, root-path, merit-dump
```
in the source this looks strange for notes. Doesn't that need an indent?

pgaultier replied:
Added the options in Notes.

fabianfrz commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+
+Create a new config
+
+.. image:: images/SFRRED_services_nginx_configuration_2.png
```
you should select the match type in the screen shot in the next line. Abusing fields may lead to unwanted side effects in future releases.

BTW: matching on "/" should do as well if you don't care about the dots.

pgaultier replied:
Well, I tried several configurations and this one is working as expected.
About the fields, this is a full screenshot because I thought it would be easier for the reader to see the whole page and know where he can leave fileds 'blank'

MichaelDeciso commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+The guide deals with internet connection and phone. Support for TV has not been tested
+
+.. Note::
+    Before starting this guide, you should have the MAC adress of your SFR/RED Box. In the guide you should replace xx:xx:xx:xx:xx:xx by your SFR/RED Box MAC adress
```
```suggestion
    Before starting this guide, you should have the MAC address of your SFR/RED Box. In this guide you should replace xx:xx:xx:xx:xx:xx with your SFR/RED Box MAC address
```

pgaultier replied:
Fixed in last commit

MichaelDeciso commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+**Getting ready to make the connection**
+----------------------------------------
+
+SFR/RED requires that the WAN interface assignment should hence look something like this
```
```suggestion
SFR/RED requires that the WAN interface assignment should look similar to this:
```

pgaultier replied:
Fixed

MichaelDeciso commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+
+Select :menuselection:`Interfaces --> [WAN]`
+
+In order to establish the IPv4 and IPv6 connection SFR/RED requires that the correct parameters are passed for the DHCP and DHCP6
```
```suggestion
In order to establish the IPv4 and IPv6 connection, SFR/RED requires that the correct parameters are passed for the DHCPv4 and DHCPv6
```

pgaultier replied:
Fixed in last commit

MichaelDeciso commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+	:width: 100%
+
+.. Note::
+    It is necessary to specify the following 'Send Options'
```
The docs use “these quotes” (or at least most of it, might have overlooked some).

pgaultier replied:
Done for the quotes - it was a copy/paste from another documentation :-)

MichaelDeciso commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+
+* domain-name-servers, domain-name
+
+Finally set the Identity Association and Prefix interface as shown
```
You'll need to be a bit clearer here. What are you specifically referring to? An earlier screenshot or table, for example?

pgaultier replied:
Adding parameters to update in Note field

MichaelDeciso commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+Interfaces / Parameters
++++++++++++++++++++++++
+
+Select :menuselection:`Interfaces --> Parameters` and set your DUID
```
You need to end the sentences with a dot. Like the previous one, I won't spam this line note. ;-)

pgaultier replied:
Should be done for all sentences

MichaelDeciso commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+
+
+.. Note::
+    It is advisable at this point to reboot the system. This will allow you to retrieve IPv4 which will be used in next part
```
```suggestion
    It is advisable at this point to reboot the system. This will allow you to retrieve an IPv4 address which will be used in next part
```
(assuming that this indeed refers to an IPv4 address and not something else, of course)

pgaultier replied:
Fixed

MichaelDeciso commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+.. Note::
+    This how-to does not cover installation of NGINX nor the use of SSH / shell commands.
+
+First login into you OPNSense firewall and create a folder **/srv/sfrredbox**. We will add in this folder the scripts used to spoof the SFR/RED Box requests.
```
```suggestion
First SSH into your OPNSense firewall and create a folder **/srv/sfrredbox**. In this folder, we will add the scripts used to spoof the SFR/RED Box requests.
```

pgaultier replied:
Fixed

MichaelDeciso commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+
+
+.. Warning::
+    Code cannot be copy / paste, you have to adjust the parameters and make it consistant with your own parameters
```
```suggestion
    Code cannot be copied and pasted as-is, you will have to adjust the parameters and make it consistent with your own settings.
```

pgaultier replied:
fixed

MichaelDeciso commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+
+
+.. Note::
+    Right now, everything should be ready. Restart the firewall, once ready plug the SFR/RED Box on you LAN and start it.
```
```suggestion
    Right now, everything should be ready. Restart the firewall, once ready plug the SFR/RED Box on your LAN and start it.
```

pgaultier replied:
Fixed

MichaelDeciso commented:
@fabianfrz I'm glad my contributions to the docs have left a good impression ^^

In the interest of full disclosure, I don't officially work at Deciso any more. However, since Deciso has been very good to me and because I like the project, I am more than happy to continue doing doc reviews.

I have just left my review and I'll happily continue to do so in the future. (Although I might switch to my other account, @Gymnasiast for that.)

@pgaultier Thanks for contributing this guide. I have left some notes to improve the wording. Besides those, make sure you look at the dots, colons (`:`) and quotation marks (`“ ”`).

pgaultier commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+-----------------
+This guide is for SFR/RED France FTTH using DHCP to connect
+
+The guide deals with internet connection and phone. Support for TV has not been tested
```
Done in last commit

pgaultier commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+
+**Introduction**
+-----------------
+This guide is for SFR/RED France FTTH using DHCP to connect
```
Merge the two paragraphs in last commit

pgaultier commented on source/manual/how-tos/sfr_red_fr_ftth.rst:
```diff
@@ -0,0 +1,286 @@
+Select :menuselection:`Interfaces --> [WAN]`
+
+In order to establish the IPv4 and IPv6 connection SFR/RED requires that the correct parameters are passed for the DHCP and DHCP6
+requests respectively
```
Added dots in last commit

pgaultier commented:
Thank you for all your remarks. I think I fixed everything.

As I don't know if I can "resolve conversation" or if you will do it, I just added comments.

fichtner commented:
@pgaultier if you feel they are resolved please mark then as such for overview purposes :)

pgaultier commented:
Done, but in my company the only one who can resolve an issue is the one who created it :). This is a process just to make sure the issues are fixed as expected.

fichtner commented:
Thanks. Well, in open source, we can't always wait for everybody :)

fichtner commented:
Merged, thanks a lot @pgaultier ❤️
